samples/csharp_dotnetcore/49.qnamaker-all-features/Dialog/QnAMakerBaseDialog.cs

```diff
@@ -63,9 +67,49 @@ protected async override Task<QnADialogResponseOptions> GetQnAResponseOptionsAsy
                 CardNoMatchText = DefaultCardNoMatchText,
                 NoAnswer = noAnswer,
                 CardNoMatchResponse = cardNoMatchResponse,
+                DisplayPreciseAnswerOnly = this.DisplayPreciseAnswerOnly
             };
 
             return responseOptions;
         }
+        private bool EnablePreciseAnser
+        {
+            get
+            {
+                var qnaServiceType = _config["QnAServiceType"];
+                if (string.Equals("v2",qnaServiceType, StringComparison.OrdinalIgnoreCase))
+                {
+                    var rawEnablePreciseAnswer = _config["EnablePreciseAnswer"];
+                    if (!string.IsNullOrWhiteSpace(rawEnablePreciseAnswer))
+                    {
+                        return bool.Parse(rawEnablePreciseAnswer);
+                    }
+                    else
+                    {
+                        return true;
+                    }
+                }
+                else
+                {
+                    return false;
+                }
+            }
+        }
+
+        private bool DisplayPreciseAnswerOnly
+        {
+            get
+            {               
+                var rawDisplayPreciseAnswerOnly = _config["DisplayPreciseAnswerOnly"];
+                if (!string.IsNullOrWhiteSpace(rawDisplayPreciseAnswerOnly))
+                {
+                    return bool.Parse(rawDisplayPreciseAnswerOnly);
+                }
+                else
```

Comment:
>else [](start = 16, length = 4)

Redundant.

Reply:
Implemented review comment.

---
In reply to: [433456660](https://github.com/microsoft/BotBuilder-Samples/pull/2430#discussion_r433456660) [](ancestors = 433456660)